File: test/aidbox_sdk/snapshots/typescript/hl7-fhir-r4-core/StructureDefinition.ts
import { UsageContext } from "./UsageContext";
import { ContactDetail } from "./ContactDetail";
import { CodeableConcept } from "./CodeableConcept";
import { Coding } from "./Coding";
import { ElementDefinition } from "./ElementDefinition";
import { DomainResource } from "./DomainResource";
import { Identifier } from "./Identifier";
import { BackboneElement } from "./BackboneElement";

export type StructureDefinitionMapping = BackboneElement & {
    uri?: string;
    name?: string;
    comment?: string;
    identity: string;
};

export type StructureDefinitionSnapshot = BackboneElement & {
    element: ElementDefinition[];
};

export type StructureDefinitionContext = BackboneElement & {
    type: string;
    expression: string;
};

export type StructureDefinitionDifferential = BackboneElement & {
    element: ElementDefinition[];
};

export type StructureDefinition = DomainResource & {
    description?: string;
    date?: string;
    derivation?: string;
    publisher?: string;
    contextInvariant?: string[];
    fhirVersion?: string;
    jurisdiction?: CodeableConcept[];
    purpose?: string;
    name: string;
    mapping?: StructureDefinitionMapping[];
    useContext?: UsageContext[];
    abstract: boolean;
    copyright?: string;
    type: string;
    experimental?: boolean;
    title?: string;
    snapshot?: StructureDefinitionSnapshot;
    keyword?: Coding[];
    status: string;
    kind: string;
    url: string;
    identifier?: Identifier[];
    context?: StructureDefinitionContext[];
    version?: string;
    differential?: StructureDefinitionDifferential;
    contact?: ContactDetail[];
    baseDefinition?: string;
};

File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/StructureDefinition.py
from pydantic import *
from typing import Optional, List
from ..base import *

class StructureDefinition_Mapping(BackboneElement):
    uri: Optional[str] = None
    name: Optional[str] = None
    comment: Optional[str] = None
    identity: str

class StructureDefinition_Snapshot(BackboneElement):
    element: list[ElementDefinition] = []

class StructureDefinition_Context(BackboneElement):
    type: str
    expression: str

class StructureDefinition_Differential(BackboneElement):
    element: list[ElementDefinition] = []

class StructureDefinition(DomainResource):
    description: Optional[str] = None
    date: Optional[str] = None
    derivation: Optional[str] = None
    publisher: Optional[str] = None
    context_invariant: Optional[List[str]] = None
    fhir_version: Optional[str] = None
    jurisdiction: Optional[List[CodeableConcept]] = None
    purpose: Optional[str] = None
    name: str
    mapping: Optional[List[StructureDefinition_Mapping]] = None
    use_context: Optional[List[UsageContext]] = None
    abstract: bool
    copyright: Optional[str] = None
    type: str
    experimental: Optional[bool] = None
    title: Optional[str] = None
    snapshot: Optional[StructureDefinition_Snapshot] = None
    keyword: Optional[List[Coding]] = None
    status: str
    kind: str
    url: str
    identifier: Optional[List[Identifier]] = None
    context: Optional[List[StructureDefinition_Context]] = None
    version: Optional[str] = None
    differential: Optional[StructureDefinition_Differential] = None
    contact: Optional[List[ContactDetail]] = None
    base_definition: Optional[str] = None